python/mooseutils/mooseutils.py

```diff
@@ -106,7 +107,22 @@ def find_moose_executable(loc, **kwargs):
     else:
         methods = ['opt', 'oprof', 'dbg', 'devel']
     methods = kwargs.pop('methods', methods)
-    name = kwargs.pop('name', os.path.basename(loc))
+    name = kwargs.pop('name', None)
+
+    # If the 'name' is not provided first look for a Makefile with 'APPLICATION_NAME...' if
+    # that is not found use the name of the directory
+    if name is None:
```

Comment:
Maybe we should make this be a loop `while name != '/'`.  You will need a break condition to for when we reach `/` and then modify the path each iteration look in the parent directory.  Then this would work even if you were in subdirectories of the repo.  We do the same thing when looking for the `testroot` file.

Reply:
there is another script for recursively looking up a moose exe, it uses this at each level

Reply:
https://github.com/idaholab/moose/blob/62c4878d7bd4e02a94b764c9900d5a57295a97a8/python/mooseutils/mooseutils.py#L137

Reply:
Cool.  Is the recursive function used in @pbalest's case?